### PR TITLE
mgmt, update monitor to composite-v1

### PR DESCRIPTION
**Pull request by XiaofeiCao**
# Description

link https://github.com/Azure/azure-sdk-for-java/issues/27617
manual code in https://github.com/Azure/azure-sdk-for-java/commit/81c116f461f3cd01b3dd3b6e87c902d51b3bc0aa, renamed back some models in https://github.com/Azure/azure-sdk-for-java/pull/30607/commits/6663d519cc1822421db273be9108c15ac4f800f4 to align with the convenience layer models.
also some test adjustments
live tests pass

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager-monitor/CHANGELOG.md [hunk 1]
```diff
@@ -2,13 +2,12 @@
...
-
 ### Breaking Changes
 
-### Bugs Fixed
-
-### Other Changes
+- Removed unused classes.
+- Removed `JsonFlatten` annotation from model classes.
+- Added `final` modifier to model classes previously annotated with `JsonFlatten`.
+- Changed type of `CategoryType` from enum to class.
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-monitor/CHANGELOG.md:10]: This is a breaking change in convenience layer, currently no idea on how to deal with it...

File: sdk/resourcemanager/azure-resourcemanager-monitor/CHANGELOG.md [hunk 2]
```diff
@@ -2,13 +2,12 @@
...
-### Features Added
-
 ### Breaking Changes
 
-### Bugs Fixed
-
-### Other Changes
+- Removed unused classes.
+- Removed `JsonFlatten` annotation from model classes.
+- Added `final` modifier to model classes previously annotated with `JsonFlatten`.
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-monitor/CHANGELOG.md:9]: no need to mention this 2

File: sdk/resourcemanager/azure-resourcemanager-monitor/CHANGELOG.md [hunk 3]
```diff
@@ -2,13 +2,12 @@
 
 ## 2.18.0-beta.1 (Unreleased)
 
-### Features Added
-
 ### Breaking Changes
 
-### Bugs Fixed
-
-### Other Changes
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-monitor/CHANGELOG.md:11]: line on dependency change?

**Review by weidongxu-microsoft — COMMENTED** [review]
I guess we will take the change on `CategoryType` (please verify that it is in GA api-version)

File: sdk/resourcemanager/api-specs.json [hunk 4]
```diff
@@ -163,7 +163,8 @@
     "dir": "azure-resourcemanager-monitor",
     "source": "specification/monitor/resource-manager/readme.md",
     "package": "com.azure.resourcemanager.monitor",
-    "args": "--tag=package-2021-04 --add-inner=AutoscaleProfile,ScaleRule,LocalizableString --rename-model=MetricUnit:Unit"
+    "args": "--tag=package-composite-v1 --add-inner=AutoscaleProfile,ScaleRule,LocalizableString,DiagnosticSettingsCategoryResourceCollection --rename-model=MetricUnit:Unit,ActionGroupAutoGenerated:ActivityLogAlertActionGroup,AlertRuleAllOfCondition:ActivityLogAlertAllOfCondition,AlertRuleAnyOfOrLeafCondition:ActivityLogAlertLeafCondition,ActionGroupAutoGenerated:ActivityLogAlertActionGroup",
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/api-specs.json:166]: why we have these AutoGenerated class? Are them new?

Comment (weidongxu-microsoft) [reply sdk/resourcemanager/api-specs.json:166]: the problem for these, are service could potential "fix" them and having a single `ActionGroup` in future.

Comment (XiaofeiCao) [reply sdk/resourcemanager/api-specs.json:166]: No, they've been renamed in the newer specs.
`ActivityLogAlertActionGroup`'s been rename to `ActionGroup`:

`ActivityLogAlertActionGroup` from `2017-04-01`:
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/monitor/resource-manager/Microsoft.Insights/stable/2017-04-01/activityLogAlerts_API.json#L437

`ActionGroup` from `2020-10-01`:
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/monitor/resource-manager/Microsoft.Insights/stable/2020-10-01/activityLogAlerts_API.json#L477

`AutoGenerated` suffix is added by m4, resolving duplicate model name from that in `actionGroups`:
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/monitor/resource-manager/Microsoft.Insights/stable/2022-06-01/actionGroups_API.json#L883

Comment (weidongxu-microsoft) [reply sdk/resourcemanager/api-specs.json:166]: wondering if there is minor difference and a few directives could merge them.

or one of them is from a preview that we do not need, and that preview json can be excluded.

Comment (weidongxu-microsoft) [reply sdk/resourcemanager/api-specs.json:166]: OK, could we add readme.java.md to do the rename?

Rely on m4 is unreliable, as one does not know which one get to `ActionGroup`, which one to `ActionGroupAutoGenerated`. That might depends on the sequence of json in tag.

Comment (XiaofeiCao) [reply sdk/resourcemanager/api-specs.json:166]: As the description implies, `ActionGroup` defined in `activityLogAlert` acts as a pointer to that from `actionGroup`:
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/monitor/resource-manager/Microsoft.Insights/stable/2020-10-01/activityLogAlerts_API.json#L478

Don't know why they renamed the model...

Comment (XiaofeiCao) [reply sdk/resourcemanager/api-specs.json:166]: > OK, could we add readme.java.md to do the rename of these duplicated models?
> 
> Rely on m4 is unreliable, as one does not know which one get to `ActionGroup`, which one to `ActionGroupAutoGenerated`. That might depends on the sequence of json in tag.

Sure!

File: sdk/resourcemanager/azure-resourcemanager-monitor/CHANGELOG.md (hunk 2 above)
Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-monitor/CHANGELOG.md]: Removed.

File: sdk/resourcemanager/azure-resourcemanager-monitor/CHANGELOG.md (hunk 3 above)
Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-monitor/CHANGELOG.md:11]: Added.

**Comment by XiaofeiCao** [comment]
Behavioral change on `enableReceiver`:

https://github.com/Azure/azure-rest-api-specs/blob/main/specification/monitor/resource-manager/Microsoft.Insights/stable/2019-06-01/actionGroups_API.json#L324

https://github.com/Azure/azure-rest-api-specs/blob/main/specification/monitor/resource-manager/Microsoft.Insights/stable/2022-06-01/actionGroups_API.json#L615

**Comment by weidongxu-microsoft** [comment]
For the 409, we can keep the previous behavior, if there is no confirmation for service.

**Review by weidongxu-microsoft — APPROVED** [review]

**Comment by XiaofeiCao** [comment]
service prefer we treat 409 as an error response

**Review by weidongxu-microsoft — APPROVED** [review]

File: sdk/resourcemanager/azure-resourcemanager/pom.xml [hunk 5]
```diff
@@ -57,6 +57,7 @@
 
       --add-opens com.azure.core/com.azure.core.implementation.util=ALL-UNNAMED
     </javaModulesSurefireArgLine>
+    <revapi.skip>true</revapi.skip>
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager/pom.xml:60]: what's the change?

this likely breaks downstream

Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager/pom.xml:60]: 1. changed `CategoryType` from enum to class
2. removed some classes
3. removed `JsonFlatten` annotation

**Review by weidongxu-microsoft — APPROVED** [review]

**Comment by weidongxu-microsoft** [comment]
you need to merge conflict and update changelog (again)

**Comment by XiaofeiCao** [comment]
Yes, I found some versions are not updated since last release. 
https://github.com/Azure/azure-sdk-for-java/pull/30664/files#diff-045e0e1c0196a5fe74796e7fabd4a5d7b2197e8ad03ac845f70ca5bdda90bece
Fixing them...

**Pull request merged**